File: bridge/java/PotatoFry.java
package bridge.java;

public class PotatoFry extends Food {
    public PotatoFry(Pan pan) {
        super(pan);
    }

    @Override
    public void prepare() {
        System.out.println("Preparing Potato Fry");
        pan.cook();
    }
}

File: bridge/golang/potatofry.go
package main

type PotatoFry struct {
	pan Pan
}

func NewPotatoFry(pan Pan) *PotatoFry {
	return &PotatoFry{pan: pan}
}

func (pf *PotatoFry) Prepare() string {
	return "Preparing potato fry with " + pf.pan.Cook()
}
